public/js/controllers/main.controller.js

```diff
@@ -11,14 +11,13 @@ angular.module('main', [])
 
         $scope.createTodo = function () {
             $scope.loading = true;
-            if ($scope.newTodo.text !== "") {
+            if ($scope.newTodo.text !== undefined) {
                 Todo.create($scope.newTodo)
                     .success(function (data) {
-                        $scope.newTodo = { text: "" };
-
                         var todos = $scope.todos;
                         todos.push(data);
 
+                        $scope.newTodo = {};
                         $scope.loading = false;
                     })
                     .error(function (data, status) { $scope.loading = false; });
```

Comment:
A `finally` callback you can set to promise like:
```javascript```
Todo.create(fun...)
        .success(fun...)
        .error(fun...)
        .finally(function () {
               $scope.loading = false;
        })

```